### PR TITLE
Working version

### PR DESCRIPTION
**Pull request by Cisfinitum**
H2 database + Spring Security + cool structure

### CODE REVIEW

File: src/main/java/com/epam/DAO/PersonService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.epam.DAO;
+
+import com.epam.model.Person;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
```
Comment (goponenkoev) [inline src/main/java/com/epam/DAO/PersonService.java]: Unuseful space between imports

File: src/main/java/com/epam/config/DBConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.epam.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+
```
Comment (goponenkoev) [inline src/main/java/com/epam/config/DBConfiguration.java]: And here

File: src/main/java/com/epam/DAO/PersonService.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.sql.ResultSet;
+import java.util.List;
+
+@Service
+public class PersonService {
```
Comment (goponenkoev) [inline src/main/java/com/epam/DAO/PersonService.java]: I think you should make sure that there is the final version of your code here. 'Cause I can find PersonService + PersonDAO

File: src/main/java/com/epam/DAO/PersonService.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public PersonService(NamedParameterJdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public List<Person> getPersons(){
+        return jdbcTemplate.query("SELECT * FROM person",
+                (ResultSet rs, int rowNum) -> {
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/DAO/PersonService.java]: We dont need to use ResultSet and int in lambdas - we can just name params.
Also, it is highly recommended to encapsulated such lambdas in methods and use method reference type of lambdas (it increases readability and testing)

File: src/main/java/com/epam/DAO/PersonService.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public PersonService(NamedParameterJdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public List<Person> getPersons(){
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/DAO/PersonService.java]: I think, getAll is better name for the method, dont you?

File: src/main/java/com/epam/controller/MainController.java [hunk 6]
```diff
@@ -1,17 +1,33 @@
...
+  private final PersonService personService;
+  @Autowired
+  public MainController(PersonService personService) {
+    this.personService = personService;
+  }
+
+
   @GetMapping("/index")
   public String defaultPage(Principal user, ModelMap model) {
+    List<Person> persons = personService.getPersons();
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/controller/MainController.java]: Maybe not persons, but people?

File: src/main/java/com/epam/controller/MainController.java [hunk 7]
```diff
@@ -1,17 +1,33 @@
...
+    for(Person person: persons){
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/controller/MainController.java]: If want to use java 8, can we use some new cool things, such as people.forEach() method. What do you think about it?

File: src/main/java/com/epam/controller/MainController.java [hunk 8]
```diff
@@ -1,17 +1,33 @@
...
+      System.out.println(person.getName());
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/controller/MainController.java]: Not sout, please, you logs

Comment (imbirevkotlya) [reply src/main/java/com/epam/controller/MainController.java]: Also, it is great chance to use toString method, why not?

File: src/main/webapp/WEB-INF/root-context.xml [hunk 9]
```diff
@@ -1,13 +1,15 @@
...
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd">
 
+    <context:component-scan base-package="com.epam" />
     <bean name="MyUserDetailsService" class="com.epam.service.UserDetailsServiceImpl">
-        <property name="userService" ref="MyUserService" />
+
     </bean>
 
-    <bean name="MyUserService" class="com.epam.service.UserServiceImpl" />
+    <!--<bean name="MyUserService" class="com.epam.service.UserServiceImpl" />-->
```
Comment (imbirevkotlya) [inline src/main/webapp/WEB-INF/root-context.xml]: Commented code is a not good option now, we have git, so we dont have to save comments for future.

File: build.gradle [hunk 10]
```diff
@@ -27,25 +27,23 @@ dependencies {
     compile group: 'org.springframework.security', name: 'spring-security-web', version: '5.1.3.RELEASE'
     compileOnly "org.projectlombok:lombok:1.18.4"
     testCompileOnly "org.projectlombok:lombok:1.18.4"
-    compile group: 'org.springframework', name: 'spring-webmvc', version: '5.1.4.RELEASE'
+    compile group: 'org.springframework', name: 'spring-webmvc', version: '5.1.3.RELEASE'
```
[the remark this answers is not in the document]
Comment (imbirevkotlya) [reply build.gradle]: Take a look at the same tomcat variable in out project. Also this article can help
https://stackoverflow.com/questions/34675914/using-variables-in-gradle-build-script

File: src/main/java/com/epam/DAO/PersonService.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
+package com.epam.DAO;
```
[the remark this answers is not in the document]
Comment (imbirevkotlya) [reply src/main/java/com/epam/DAO/PersonService.java]: https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html

[the remark this answers is not in the document]
Comment (imbirevkotlya) [reply src/main/java/com/epam/DAO/PersonService.java]: https://www.oracle.com/technetwork/java/codeconventions-150003.pdf

File: src/main/java/com/epam/DAO/PersonService.java (hunk 4 above)
Comment (imbirevkotlya) [reply src/main/java/com/epam/DAO/PersonService.java]: https://habr.com/ru/post/224593/

Comment (imbirevkotlya) [reply src/main/java/com/epam/DAO/PersonService.java]: https://www.youtube.com/watch?v=7Iy1hVEXxsU

File: src/main/java/com/epam/controller/MainController.java (hunk 7 above)
Comment (imbirevkotlya) [reply src/main/java/com/epam/controller/MainController.java]: https://habr.com/ru/post/337350/

File: src/main/webapp/WEB-INF/root-context.xml (hunk 9 above)
Comment (imbirevkotlya) [reply src/main/webapp/WEB-INF/root-context.xml]: http://cat-in-web.ru/wp-content/uploads/books/martin-chistyj-kod.pdf 
глава 4

**Pull request closed without merging**